src/core/a-entity.js

```diff
@@ -594,16 +610,20 @@ var proto = Object.create(ANode.prototype, {
    */
   setAttribute: {
     value: function (attr, value, componentPropValue) {
-      var isDebugMode = this.sceneEl && this.sceneEl.getAttribute('debug');
-      var componentName = attr.split(MULTIPLE_COMPONENT_DELIMITER)[0];
+      var componentName;
+      var isDebugMode;
+
+      componentName = attr.split(MULTIPLE_COMPONENT_DELIMITER)[0];
       if (COMPONENTS[componentName]) {
-        // Just update one of the component properties
+        // Just update one of the component properties.
         if (typeof value === 'string' && componentPropValue !== undefined) {
           this.updateComponentProperty(attr, value, componentPropValue);
         } else {
           this.updateComponent(attr, value);
         }
-        // On debug mode we write the component state to the DOM attributes
+
+        // On debug mode we write the component state to the DOM attributes.
+        isDebugMode = this.sceneEl && this.sceneEl.getAttribute('debug');
```

Comment:
I would leave the variables at top. If you want to reuse the same variable somewhere else in the function you have to remember to move it to the top or at least above the other place you want to use it.